flight_computer/lib/opt3001.py

```diff
@@ -0,0 +1,156 @@
+import time
+
+from adafruit_bus_device.i2c_device import I2CDevice
+
+# Can be one of four possible values
+# This assumes we connect the ADDR pin to GND
+# See 8.3.4.1 in the specification
+DEVICE_DEFAULT_I2C_ADDR = 0x44
+
+# MAIN REGISTERS FOR PROGRAMMING DEVICE
+RESULT = 0x00
+CONFIGURATION = 0x1
+
+# CONFIGURATION SETTINGS
+AUTO_RANGE_SETTING = 0b1100
+CONTINUOUS_CONVERSIONS = 0b11
+DEFAULT_CONVERSION_TIME = 800
+MAX_CONVERSION_TIME = 800
+
+# FOR INTERRUPT MECHANISMS, UNUSED
+LOW_LIMIT = 0x2
+HIGH_LIMIT = 0x3
+
+# CAN BE READ TO GET ID
+MANUFACTURER_ID = 0x7E
+DEVICE_ID = 0x7F
+
+REGISTER_SIZE = 2
+
+
+# EXTRACT k bits from and including pos p
+# using n ... 0 indexing
+def extract_bits(bits, pos, k):
+    # first chop off everything exclusively after pos
+    # then get k inclusive least significant bits from the 0 index
+    return (bits >> pos) & ((1 << k) - 1)
+
+
+class Opt3001:
+    """OPT3001 Sun Sensor Driver"""
+
+    def __init__(self, i2c_bus, address=DEVICE_DEFAULT_I2C_ADDR, conversion_time=DEFAULT_CONVERSION_TIME):
+        # Initialize and Configure the sun sensor
+
+        # Initialize i2c device from i2c passed in
+        self.i2c_device = I2CDevice(i2c_bus, address)
+
+        # Initialize a fixed buffer to read and write from
+        self.buf = bytearray(3)
+
+        # DEVICE ID SHOULD BE 0x3001
+        self.read_value(DEVICE_ID)
+        if self.buf != bytes([DEVICE_ID, 0x30, 0x01]):
```

Comment:
It might be better to either slice the buffer, i.e. self.buf[2:3] or maybe we should just make the buffer two bytes wide and overwrite the register.